### PR TITLE
Multiselect > programmatically select by row ids

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #37 

Added `selectedRowIds` input property to grids.

Docs update needed: blackbaud/skyux2-docs#574

### CODE REVIEW

File: src/app/public/modules/grid/grid.component.ts [hunk 1]
```diff
@@ -167,6 +167,19 @@ export class SkyGridComponent implements OnInit, AfterContentInit, OnChanges, On
...
+  @Input()
+  set multiselectSelectedRowIds(value: Array<string>) {
+    for (let i = 0; i < this.items.length; i++) {
+      this.items[i].isSelected = (value.indexOf(this.items[i].id) > -1);
+    }
+    this._multiselectSelectedRowIds = value;
+    this.ref.markForCheck();
+  }
+
+  get multiselectSelectedRowIds() {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: Needs access modifier `public`.
Needs return value.

File: src/app/public/modules/grid/grid.component.ts [hunk 2]
```diff
@@ -167,6 +167,19 @@ export class SkyGridComponent implements OnInit, AfterContentInit, OnChanges, On
   @Input()
   public multiselectRowId: string;
 
+  @Input()
+  set multiselectSelectedRowIds(value: Array<string>) {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: Needs access modifier.

File: src/app/public/modules/grid/grid.component.ts [hunk 3]
```diff
@@ -167,6 +167,19 @@ export class SkyGridComponent implements OnInit, AfterContentInit, OnChanges, On
...
+    for (let i = 0; i < this.items.length; i++) {
+      this.items[i].isSelected = (value.indexOf(this.items[i].id) > -1);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: What happens if the `value` is undefined?

File: src/app/public/modules/grid/grid.component.ts [hunk 4]
```diff
@@ -167,6 +167,19 @@ export class SkyGridComponent implements OnInit, AfterContentInit, OnChanges, On
...
+    }
+    this._multiselectSelectedRowIds = value;
+    this.ref.markForCheck();
+  }
+
+  get multiselectSelectedRowIds() {
+    return this._multiselectSelectedRowIds;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: Should we return a default value of `[]` if `this._multiselectSelectedRowIds` is undefined?

File: src/app/public/modules/grid/grid.component.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Fixed.

File: src/app/public/modules/grid/grid.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Fixed.

File: src/app/public/modules/grid/grid.component.ts (hunk 3 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: I've added a check in here to protect from this. In this case, no rows are selected and no errors are thrown.

File: src/app/public/modules/grid/grid.component.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Indeed. I've updated this to do just that.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Actually, there's no need for consumers to get this value. I've removed the getter completely.

**Pull request merged**